documentation/handling_failures.md

```diff
@@ -55,3 +55,56 @@ In the future, this step may be performed by the workflow, automatically moving
     ```
 Then call the Reaper `Reaper --kill-list kill-list.json` [Ref: https://github.com/ICGC-TCGA-PanCancer/pancancer_launcher]
 The instances will be wiped out and new workers can be provisioned to take their place.
+
+### Automatic Worker Instance Termination
+
+The production system automatically terminates worker instances after the workflow it was running failed. The stdout and stderr is logged along with the event in the launcher's database. However, the s3-transfer-operations repos is not automatically updated to indicate the occurence of failure.
+
+A script was created to compare the failed job record in the database with the location of the corresponding JSON job description file in s3-transfer-operations. In addition to locating the file, it will also try to determine a reason for failure by searching for keywords in the stdout/stderr logs to provide a more descriptive commit message.
+
+The script currently has execution of `git mv` and `git commit` instructions disabled for safety reasons, the `-os` flag should be used to output them to the console for examination.
```

Comment:
I think this will be fine for long-term (so not just currently). Output the git commands, allow a user to stage the changes in git, and then push on their own.